sdk/core/core/inc/az_action.h

```diff
@@ -40,6 +37,15 @@ AZ_STATIC_ASSERT(sizeof(size_t) == sizeof(void *))
  * the first argument of the function @NAME and @ACTION.self.
  */
 #define AZ_ACTION_FUNC(NAME, SELF, ACTION) \
+  AZ_NODISCARD az_result NAME(SELF, AZ_ACTION_ARG(ACTION) const, SELF * out); \
+  AZ_NODISCARD AZ_INLINE ACTION AZ_CAT(NAME, _action)(SELF * const self) { \
+    return (ACTION){ \
+      .func = (az_result(*)(void *, AZ_ACTION_ARG(ACTION)))NAME, \
+      .self = (void *)(size_t)self, \
+    }; \
+  }
+
+#define AZ_ACTION_FUNC_2(NAME, SELF, ACTION) \
```

Comment:
Is this temporary until we remove AZ_ACTION?

Reply:
yes, I am currently working with @JeffreyRichter  to remove Actions.